pkg/labels/fake_applicator.go

```diff
@@ -102,21 +105,73 @@ func (app *fakeApplicator) GetMatches(selector labels.Selector, labelType Type)
 	return results, nil
 }
 
-func (app *fakeApplicator) WatchMatches(selector labels.Selector, labelType Type, quitCh chan struct{}) chan []Labeled {
+func (app *fakeApplicator) WatchMatches(selector labels.Selector, labelType Type, quitCh <-chan struct{}) chan []Labeled {
 	ch := make(chan []Labeled)
+	timer := time.NewTimer(time.Duration(0))
 	go func() {
 		for {
+			select {
+			case <-quitCh:
+				return
+			case <-timer.C:
+			}
+			// upper bound on request rate
```

Comment:
We won't need to worry about it for performance, but I'll need it so the changes are all captured together for the tests to work consistently.